File: src/main/clients/openrgb/effects/SpectrumCycle.ts
import { RGBColor } from '@main/clients/openrgb/client/classes/RGBColor';
import { AbstractEffect } from '@main/clients/openrgb/effects/base/AbstractEffect';

class SpectrumCycle extends AbstractEffect {
  public constructor() {
    super('Spectrum Cycle', false);
  }

  protected async applyEffect(): Promise<void> {
    for (let offset = 0; this.isRunning; offset = (offset + 1) % 360) {
      this.devices.forEach((dev) => {
        this.setColors(dev, Array(dev.leds.length).fill(RGBColor.fromHSV(offset, 1, 1)));
      });
      await this.sleep(20);
    }
    this.hasFinished = true;
  }
}

export const spectrumCycle = new SpectrumCycle();

File: lib/clients/openrgb/effects/spectrum_cycle.py
from .base.abstract_effect import AbstractEffect
from ..sdk.utils import RGBColor
from ....utils.singleton import singleton

import time

@singleton
class SpectrumCycle(AbstractEffect):
    def __init__(self):
        super().__init__("Spectrum cycle")

    def apply_effect(self):
        offset = 0
        while self.is_running:
            for dev in self.devices:
                colors = [RGBColor.fromHSV(offset, 1, 1)] * len(dev.leds)
                self.set_colors(dev, colors)
            offset = (offset + 1) % 360
            self.sleep(0.02)  # 20 ms

spectrum_cycle = SpectrumCycle()